project-management-openproject/project-management-openproject-api/src/main/java/com/xwiki/projectmanagement/openproject/internal/OpenProjectClient.java

```diff
@@ -194,16 +196,24 @@ private String extractProjectFromPath(String path)
         return matcher.find() ? matcher.group(1) : null;
     }
 
-    private String extractFiltersFromQuery(String filtersString, List<LiveDataQuery.Filter> filtersList)
-        throws ProjectManagementException
+    private String extractFiltersFromQuery(JsonNode filtersNode, List<LiveDataQuery.Filter> filtersList)
+        throws ProjectManagementException, JsonProcessingException
     {
-        return OpenProjectFilterHandler.mergeFilters(filtersList, filtersString);
+        if (filtersNode != null) {
+            String filtersString = objectMapper.writeValueAsString(filtersNode);
+            return OpenProjectFilterHandler.mergeFilters(filtersList, filtersString);
```

Comment:
I dont think i like this. The filters are extracted from the url as string and then parsed into a JsonNode. The jsonNode is then transformed into a string again (right here) just to be transformed into a jsonNode again later.